client/src/components/TodoTable.tsx

```diff
@@ -11,68 +12,149 @@ import {
   Toolbar,
   Typography,
 } from "@mui/material";
-import { TABLE_HEADERS } from "../constants/constants";
+import { TABLE_HEADERS, TODO_TYPES } from "../constants/constants";
 import useGetTodosQuery from "../api/todo_service/getTodos";
 import TodoRows from "./TodoRows";
 import { useSearchParams } from "react-router-dom";
 import { InputFormModal } from "./InputFormModal";
-import { useEffect } from "react";
+import { useEffect, useMemo, useState } from "react";
 import TodosTablePagination from "./TablePagination";
 import useDeleteTodoMutation from "../api/todo_service/deleteTodo";
+import TodoSearchBar from "./TodoSearch";
+import { AxiosError, HttpStatusCode } from "axios";
+import TodoFilterMenu from "./TodoFilter";
 
 const TodosTable = () => {
-  const [searchParams] = useSearchParams();
+  const [searchParams, setSearchParams] = useSearchParams();
   const page = Number(searchParams.get("page") || "1");
-  const { data, refetch } = useGetTodosQuery(page);
+  const [searchTerm, setSearchTerm] = useState(searchParams.get("name") || "");
+  const [types, setTypes] = useState<string[]>(
+    (() => {
+      const typeParam = searchParams.get("type");
+      return typeParam === "[]"
+        ? TODO_TYPES
+        : typeParam
+        ? (() => {
```

Comment:
I see that you're using that function twice per code. Can you move it to the helper? Let's call it like ParseTodoTypes?